scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatWriter.scala

```diff
@@ -594,18 +595,42 @@ class FormatWriter(formatOps: FormatOps) {
           if (matches == 0 || doubleNewline || !locationIter.hasNext) {
             var column = 0
             val columns = minMatches
+
+            /**
+              * Separator length gap needed for align blocks with different token
+              * lengths by expression names, not tokens itself.
+              *
+              * Without gaps considering:
+              * ```
+              * libraryDependencies ++= Seq(
+              *   "org.scalacheck"  %% "scalacheck" % scalacheckV,
+              *   "io.get-coursier" % "interface"   % "0.0.17"
+              * )
+              * ```
+              *
+              * With gaps considering:
+              * ```
+              * libraryDependencies ++= Seq(
+              *   "org.scalacheck" %% "scalacheck" % scalacheckV,
+              *   "io.get-coursier" % "interface"  % "0.0.17"
+              * )
+              * ```
+              * */
+            val previousSeparatorLengthGaps = Array.fill(block.length)(0)
             while (column < columns) {
-              val blockWithWidth = block.map { line =>
-                val location = line(column)
-                val previousWidth =
-                  if (column == 0) 0 else line(column - 1).shift
-                val key = location.shift - previousWidth
-                key -> hash(location.formatToken.left)
-              }
-              val maxWidth = blockWithWidth.map(_._1).max
-              blockWithWidth.foreach {
-                case (width, tokenHash) =>
-                  finalResult += tokenHash -> (maxWidth - width)
+              val alignmentUnits =
+                prepareAlignmentInfo(block, previousSeparatorLengthGaps, column)
+
+              val maxWidth = alignmentUnits.map(_.width).max
+
+              val maxWidthTokenLength =
+                alignmentUnits.maxBy(_.width).separatorLength
+
+              alignmentUnits.foreach { info =>
```

Comment:
why not `alignmentUnits.zipWithIndex.foreach` and remove the `lineIndex` field below?

Reply:
Because I want protection against further code changes which may broke `alignmentUnits` order. `lineIndex` field always contain actual line number in origin array